src/megaclient.cpp

```diff
@@ -2600,7 +2600,12 @@ bool MegaClient::dispatch(direction_t d)
 
                 if (d == PUT)
                 {
-                    nexttransfer->size = ts->fa->size;
+                    if (ts->fa->mtime != nexttransfer->mtime || ts->fa->size != nexttransfer->size)
+                    {
+                        LOG_warn << "Modification detected starting upload";
```

Comment:
Minor issue:
"Modification detected" is warned twice for changes detected on startup (one for starting & one "during"):
15:59:47 (warn):  Modification detected starting upload (megaclient.cpp:2601)
15:59:47 (debug): Transfer failed with error -21 (transfer.cpp:359)
16:00:24 (warn):  Modification detected during active upload (transfer.cpp:395)